### PR TITLE
fix: add correlate2d/convolve2d aliases for filter2d (#2298)

### PR DESCRIPTION
**Pull request by fumadari**
## 📝 Description

**⚠️ Issue Link Required**: This PR must be linked to an approved and assigned issue. See [Contributing Guide](https://github.com/kornia/kornia/blob/main/CONTRIBUTING.md#pull-request) for details.

**Fixes/Relates to:** #2298

**Important**:
- Ensure you are assigned to the linked issue before submitting this PR
- This PR should strictly implement what the linked issue describes
- Do not include changes beyond the scope of the linked issue

---

## 🛠️ Changes Made
- [x] Added `correlate2d` and `convolve2d` wrappers in `kornia/filters/filter.py` that delegate to `filter2d` with fixed `behaviour`.
- [x] Exported both functions from `kornia/filters/__init__.py`.
- [x] Added tests in `tests/filters/test_filters.py` to verify alias equivalence and the kernel-flip relationship between convolution and correlation.
- [x] Updated docs in `docs/source/filters.rst` to include both new functions in the Filtering API section.

---

## 🧪 How Was This Tested?
- [x] **Unit Tests:** Added `test_correlation_and_convolution_aliases` and `test_convolve2d_kernel_flip_relationship`.
- [x] **Manual Verification:** Verified wrappers map exactly to `filter2d(..., behaviour="corr"|"conv")`.
- [x] **Performance/Edge Cases:** Covered both `same` and `valid` padding modes, and normalized/non-normalized modes.

Local test logs:

```bash
uv run --extra dev ruff check kornia/filters/filter.py kornia/filters/__init__.py tests/filters/test_filters.py
# All checks passed!

uv run --extra dev pytest tests/filters/test_filters.py -v
# 66 passed, 2 skipped
```

Reference implementation for behavior semantics:
- SciPy `correlate2d`: https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.correlate2d.html
- PyTorch `conv2d`: https://pytorch.org/docs/stable/generated/torch.nn.functional.conv2d.html

---

## 🕵️ AI Usage Disclosure
*Check one of the following:*
- [ ] 🟢 **No AI used.**
- [ ] 🟡 **AI-assisted:** I used AI for boilerplate/refactoring but have manually reviewed and tested every line.
- [x] 🔴 **AI-generated:** (Note: These PRs may be subject to stricter scrutiny or immediate closure if the logic is not explained).

---

## 🚦 Checklist
- [ ] I am assigned to the linked issue (required before PR submission)
- [x] The linked issue has been approved by a maintainer
- [x] I have performed a **self-review** of my code (no "ghost" variables or hallucinations).
- [x] My code follows the existing style guidelines of this project.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] (Optional) I have attached screenshots/recordings for UI changes.

---

## 💭 Additional Context
I already commented on #2298 with the implementation plan and requested assignment. This PR keeps the change minimal and focused on discoverability aliases only.

### CODE REVIEW

**Comment by fumadari** [comment]
Note on `pre-commit.ci` failure:

- Running pre-commit only on changed files passes locally:

```bash
uv run --extra dev pre-commit run --files \
  kornia/filters/filter.py \
  kornia/filters/__init__.py \
  tests/filters/test_filters.py \
  docs/source/filters.rst
# all hooks passed
```

- The failing hook reported by pre-commit.ci is `UP042` in `kornia/config.py` (`InstallationMode(str, Enum)`), which is outside this PR diff.

**Comment by fumadari** [comment]
Re-triggered CI with commit 3e3444fd due tutorial job failure in notebook execution ( image read mismatch), which appears unrelated to this diff.

**Comment by fumadari** [comment]
Follow-up (correcting previous formatting):

I retriggered CI with commit 3e3444fd because the tutorials job failed while executing notebook `_nbs/connected_components.ipynb` with an image read mismatch, which appears unrelated to this PR diff.

**Comment by fumadari** [comment]
Additional context on the tutorials failure:

I checked another recent PR run (`#3557`, run 22106131103) and it also fails in `tutorials / tutorials (3.12)` at the same workflow stage (`Run tutorials`), which suggests this is not specific to this change.

I can rerun again if maintainers prefer, but I wanted to document that pattern here.

**Comment by dario-fumarola** [comment]
Migrated to https://github.com/kornia/kornia/pull/3578 from @dario-fumarola for account consolidation. Please continue review there.

**Comment by dario-fumarola** [comment]
Superseded by https://github.com/kornia/kornia/pull/3578

**Comment by fumadari** [comment]
Superseded by https://github.com/kornia/kornia/pull/3578
